src/doc/contrib/src/tests/writing.md

```diff
@@ -64,6 +64,21 @@ test.
   - See [`support::compare`] for an explanation of the string pattern matching.
     Patterns are used to make it easier to match against the expected output.
 
+#### Filesystem layout testing
+
+Tests often to need to verify Cargo created/removed files.
+The `CargoPathExt` trait (implmented by `Path` and `PathBuf`) provides a `assert_file_layout()` to verify a file system tree against a snapshot. The snapshot format is very similar to the unix `tree` command output.
+
+Files vary across operating systems, for example `.pdb` files on Windows. 
```

Comment:
I wonder instead this platform specific snapshot we should just have different snapshots for different platforms. 

Or maybe we do this kind of tree layout test assertions only on one platform, presumably Linux, which is the most available platforms.
For example, I don't have a Windows physical machine. Some don't have macbook. But everyone can access a Linux container :)

Also, some tier 2 and 3 platform maintainers are running Cargo tests in their CI. While they already patched out some tests, we may want to be nice not to fail their CI more.

Reply:
Yeah that is why another alternative is testing only on Linux. Anything we really want to test platform differences here?